### PR TITLE
Change use of Router.Helpers to VerifiedRoutes

### PR DESCRIPTION
**Pull request by liamstevens111**
https://github.com/nimblehq/elixir-templates/issues/??

## What happened 👀

Using dynamic paths with `p sigil` : https://elixirforum.com/t/phoenix-1-7-0-rc-0-is-out/51625/21

I attempted similiar but the `#` character got encoded to %23 in the path, ie:
```elixir
"xlink:href": ~p"/images/#{"icon-sprite.svg#" <> @svg_shape_id_prefix <> icon_file_name}"
```

So I left it as:

```elixir
"xlink:href": ~p"/images/icon-sprite.svg#" <> "#{@svg_shape_id_prefix}#{icon_file_name}"
```

There is only compile warnings that the route doesn't exist if the path `images` is removed from `static_paths` in web module `(def static_paths, do: ~w(assets fonts images favicon.ico robots.txt)`

## Insight 📝

`N/A`

## Proof Of Work 📹

1. Added two svgs (arrow and ball) to `priv/static/images/icons` 
2.  ran `npm run svg-sprite.generate-icon --prefix assets` to generate `priv/static/images/icon-sprite.svg`
3. Added following code in template:

```elixir
<%= icon_tag("arrow") %>
<%= icon_tag("ball") %>
```

Outcome:

Screenshot 2023-03-15 at 14 48 12

### CODE REVIEW

File: lib/nimble_template/addons/variants/phoenix/web/svg_sprite.ex [hunk 1]
```diff
@@ -5,115 +5,109 @@ defmodule NimbleTemplate.Addons.Phoenix.Web.SvgSprite do
...
+    )
+
+    Generator.replace_content!(
+      "assets/package.json",
+      """
+        "devDependencies": {
+      """,
+      """
+        "devDependencies": {
+          "svg-sprite": "1.5.4",
```
Comment (andyduong1920) [inline lib/nimble_template/addons/variants/phoenix/web/svg_sprite.ex:45]: can we take this chance to upgrade the `svg-sprite` to `2.0.2` @liamstevens111 :D if no, we can do it on a separate PR ;)

File: priv/templates/nimble_template/lib/otp_app_web/helpers/icon_helper.ex.eex [hunk 2]
```diff
@@ -5,20 +5,17 @@ defmodule <%= web_module %>.IconHelper do
...
     classes = "icon " <> Keyword.get(opts, :class, "")
 
     content_tag(:svg, class: classes) do
       tag(:use,
         "xlink:href":
-          Routes.static_path(
-            conn,
-            "/images/icon-sprite.svg#" <> @svg_shape_id_prefix <> icon_file_name
-          )
+        ~p"/images/icon-sprite.svg#" <> "#{@svg_shape_id_prefix}#{icon_file_name}"
```
Comment (andyduong1920) [inline priv/templates/nimble_template/lib/otp_app_web/helpers/icon_helper.ex.eex:18]: Follow up with your concern `Not sure there’s any benefit using verifiedroute for static dynamic generated paths?`

Can we just use the `String` here @liamstevens111 

```suggestion
        "/images/icon-sprite.svg#" <> "#{@svg_shape_id_prefix}#{icon_file_name}"
```

Have a test on your local to see if it could render the SVG icon or not 👌

File: lib/nimble_template/addons/variants/phoenix/web/svg_sprite.ex (hunk 1 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/web/svg_sprite.ex:45]: nevermind, let's do this as a separate task, I will mark this as resolved

File: priv/templates/nimble_template/test/support/feature_case.ex.eex [hunk 3]
```diff
@@ -10,6 +10,8 @@ defmodule <%= web_module %>.FeatureCase do
       use Mimic
       use <%= web_module %>, :verified_routes
 
+      use <%= web_module %>, :verified_routes
```
Comment (andyduong1920) [inline priv/templates/nimble_template/test/support/feature_case.ex.eex]: seem we have the `use <%= web_module %>, :verified_routes` above @liamstevens111, I have added it on my PR :D 
```suggestion
```

File: test/nimble_template/addons/variants/phoenix/web/wallaby_test.exs [hunk 4]
```diff
@@ -25,6 +25,8 @@ defmodule NimbleTemplate.Addons.Phoenix.Web.WallabyTest do
                        use Mimic
                        use NimbleTemplateWeb, :verified_routes
 
+                       use NimbleTemplateWeb, :verified_routes
```
Comment (andyduong1920) [inline test/nimble_template/addons/variants/phoenix/web/wallaby_test.exs]: duplicated :)

Comment (liamstevens111) [reply test/nimble_template/addons/variants/phoenix/web/wallaby_test.exs]: 😡

File: priv/templates/nimble_template/test/otp_app_web/helpers/icon_helper_test.exs.eex [hunk 5]
```diff
@@ -5,17 +5,13 @@ defmodule <%= web_module %>.IconHelperTest do
...
   alias <%= web_module %>.IconHelper
 
-  describe "icon_tag/3" do
+  describe "icon_tag/2" do
     test "renders a svg icon" do
       first_svg_icon =
-        <%= web_module %>.Endpoint
-        |> IconHelper.icon_tag("active", class: "customize-icon-class")
-        |> safe_to_string()
+        safe_to_string(IconHelper.icon_tag("active", class: "customize-icon-class"))
```
Comment (andyduong1920) [inline priv/templates/nimble_template/test/otp_app_web/helpers/icon_helper_test.exs.eex]: can we 

```suggestion
"active"
|> IconHelper.icon_tag(class: "customize-icon-class")
       |> safe_to_string()
```

File: priv/templates/nimble_template/test/otp_app_web/helpers/icon_helper_test.exs.eex [hunk 6]
```diff
@@ -5,17 +5,13 @@ defmodule <%= web_module %>.IconHelperTest do
...
 
       second_svg_icon =
-        <%= web_module %>.Endpoint
-        |> IconHelper.icon_tag("icon-lock")
-        |> safe_to_string()
+        safe_to_string(IconHelper.icon_tag("icon-lock"))
```
Comment (andyduong1920) [inline priv/templates/nimble_template/test/otp_app_web/helpers/icon_helper_test.exs.eex]: same comment above, let's use the elixir pipe operation 💟

File: priv/templates/nimble_template/lib/otp_app_web/helpers/icon_helper.ex.eex (hunk 2 above)
Comment (liamstevens111) [reply priv/templates/nimble_template/lib/otp_app_web/helpers/icon_helper.ex.eex:18]: Yeah it renders and gives warnings for `/images` if it's not in static path, that's it. Maybe look in future to redefining it as 

~p"/images/#{"icon-sprite.svg#" <> @svg_shape_id_prefix <> icon_file_name}"

But the `#` gets encoded as `%23`

**Review by Nihisil — APPROVED** [review]

**Review by andyduong1920 — APPROVED** [review]
Thanks @liamstevens111
